### PR TITLE
(dnn) LSTM from ONNX

### PR DESCRIPTION
**Pull request by dkurt**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/730

related: https://github.com/opencv/opencv/issues/12533
related: https://github.com/opencv/opencv/issues/16662

* Slice over constant values
* `ConstantFill` and `ConstantOfShape`
* `LSTM` (not bidirectional, with zeroed initial states `h0` and `c0`)
* Squeeze in case of no ones in dimensions is Identity
* Gather without axis

<cut/>

```python
class LSTM(nn.Module):

    def __init__(self, features, hidden, batch, num_layers=1):
        super(LSTM, self).__init__()
        self.lstm = nn.LSTM(features, hidden, num_layers)
        self.h0 = torch.zeros(num_layers, batch, hidden)
        self.c0 = torch.zeros(num_layers, batch, hidden)

    def forward(self, x):
        return self.lstm(x, (self.h0, self.c0))[0]

batch = 5
features = 4
hidden = 3
seq_len = 2

input = Variable(torch.randn(seq_len, batch, features))
lstm = LSTM(features, hidden, batch)
```

```
allow_multiple_commits=1
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.1.0:16.04
build_image:Custom Win=openvino-2020.1.0
build_image:Custom Mac=openvino-2020.1.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by dkurt** [comment]
@alalek, please do not merge. There are rebase artifacts
